File: 프로그래머스/unrated/181887. 홀수 vs 짝수/홀수 vs 짝수.py
def solution(num_list):
    # 방법 2 - 슬라이싱
    return max(sum(num_list[::2]), sum(num_list[1::2]))

#     # 방법 1 - for문
#     odd, even = 0, 0
    
#     for i in range(len(num_list)):
#         if i % 2: # 인덱스가 홀수일 때, 짝수번째
#             even += num_list[i]
#         else:
#             odd += num_list[i]
    
#     return odd if odd > even else even

File: 프로그래머스/unrated/181887. 홀수 vs 짝수/홀수 vs 짝수.js
function solution(num_list) {
    // 방법 2 - filter, reduce 함수
    const odd = num_list.filter((el, i) => !(i % 2)).reduce((acc, el) => acc += el, 0)
    const even = num_list.filter((el, i) => i % 2).reduce((acc, el) => acc += el, 0)
    return odd > even ? odd : even
    
//     // 방법 1 - for문
//     let odd = 0
//     let even = 0
    
//     for (let i = 0; i < num_list.length; i++) {
//         i % 2 ? even += num_list[i] : odd += num_list[i]
//     }
    
//     return odd > even ? odd : even
}